core/arch/arm/kernel/user_ta.c

```diff
@@ -313,6 +313,14 @@ static TEE_Result ta_load(const TEE_UUID *uuid,
 		goto error_return;
 	}
 
+	if (ta_head->flags & TA_FLAG_MULTI_SESSION &&
```

Comment:
Given this I think we should not modify any flags, because doing so is a bit confusing. I'd rather interpret the flags correctly, in this case it's the if at the end of `tee_ta_close_session()` that need to take `TA_FLAG_SINGLE_INSTANCE` into account also.